File: WinTexRFID/ChoiceNumberForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace UHFAPP
	Partial Public Class ChoiceNumberForm
		Inherits Form

		Public number As Integer = 0
		Public Sub New()
			InitializeComponent()
			comboBox1.SelectedIndex = 0

			If Common.isEnglish Then

				button1.Text = "OK"
				label1.Text = "Number:"

			End If
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			If Not Me.button1.IsHandleCreated Then Return

			number = Integer.Parse(comboBox1.Text) 'comboBox1.SelectedIndex;
			Me.Close()
		End Sub

		Private Sub comboBox1_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBox1.TextChanged
			If Not Me.comboBox1.IsHandleCreated Then Return

			Dim txt As ComboBox = DirectCast(sender, ComboBox)
			If Not StringUtils.IsNumber(txt.Text) Then
				txt.Text = "0"
			Else
				If Integer.Parse(txt.Text) > 15 Then

					txt.Text = "15"
				End If

			End If

		End Sub
	End Class
End Namespace

File: WinTexRFID/Backup/ChoiceNumberForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace UHFAPP
{
    public partial class ChoiceNumberForm : Form
    {
        public int number = 0;
        public ChoiceNumberForm()
        {
            InitializeComponent();
            comboBox1.SelectedIndex = 0;

            if (Common.isEnglish) {

                button1.Text = "OK";
                label1.Text = "Number:";
            
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            number = int.Parse(comboBox1.Text); //comboBox1.SelectedIndex;
            this.Close();
        }

        private void comboBox1_TextChanged(object sender, EventArgs e)
        {
            ComboBox txt = (ComboBox)sender;
            if (!StringUtils.IsNumber(txt.Text))
            {
                txt.Text = "0";
            }
            else {
                if (int.Parse(txt.Text) > 15) {

                    txt.Text = "15";
                }
            
            }
            
        }
    }
}
